V1 design/sidebar layout

Da-Colon opened the pull request:
## Description

Adds Side bar links, tooltips and testids

## Notes

- A future PR will hide DAO links when not viewing a DAO. I will also update the Routes with the correct contract address at that time.

## Issue / Notion doc (if applicable)
Closes #542 
<!-----------------------------------------------------------------------------
If applicable, link to the relevant Github issue(s) with `closes #XXX` to auto-close it when this PR is merged.

If there is an accompanying Notion document, please link that here as well.
------------------------------------------------------------------------------>

## Testing

<!-----------------------------------------------------------------------------
If your testing steps are technical, outline steps for an engineer to verify.

If your testing steps are functional, please provide a full guide with any features requiring special attention for operations to test your branch in the preview environment.

New feature work should include a correlating automated integration test via Playwright, in collaboration with the QA team. See this repo's README.md for Playwright setup.
------------------------------------------------------------------------------>

## Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/38386583/196829158-6176f2f8-e9fc-47bf-a54d-88f16ea4f5a2.png)

![Fractal 🔊 2022-10-19 20-30-04](https://user-images.githubusercontent.com/38386583/196829282-d189163e-c82c-4efa-a24d-43f1f6357c1d.png)

Da-Colon commented on package.json:
```diff
@@ -3,7 +3,7 @@
   "dependencies": {
     "@chakra-ui/react": "^2.3.4",
-    "@decent-org/fractal-ui": "^0.0.6",
+    "@decent-org/fractal-ui": "^0.0.7",
```
v++

Da-Colon commented on src/App.tsx:
```diff
@@ -24,18 +25,16 @@ function App() {
       "nav main"`}
       gridTemplateColumns={'4.25rem 1fr'}
-      gridTemplateRows={'1fr'}
+      gridTemplateRows={['minmax(4rem, 100%) repeat(2, calc(100% - 100% +))', '100vh']}
```
This will probably change once work is done on main content and mobile designs are released.

But this is how breakpoints are handled in chakra-ui, using arrays, or if you can use the names of custom breakpoints. This will be added soon in the next release. see https://chakra-ui.com/docs/styled-system/responsive-styles#customizing-breakpoints

Da-Colon commented on src/components/ui/Sidebar/index.tsx:
```diff
@@ -0,0 +1,161 @@
+      <Flex direction="column">
+        <SidebarTooltipWrapper label="Home">
+          <Link
+            to={DAO_ROUTES.daoHome('0x')}
```
These will be updated in a future PR with a DAO address from Fractal Provider, other wise if it is not set. These links will be hidden.

Da-Colon commented on src/components/ui/Sidebar/index.tsx:
```diff
@@ -0,0 +1,161 @@
+        <SidebarTooltipWrapper label="Proposals">
+          <Link
+            to={DAO_ROUTES.proposals('0x')}
+            data-testid="sidebar-proposalsLink"
```
testids!

Da-Colon commented on src/components/ui/Sidebar/index.tsx:
```diff
@@ -0,0 +1,157 @@
+          >
+            <Proposals
+              aria-label="Proposals Link"
+              boxSize="1.5rem"
```
boxSize property basically sets the width/height of icon. see https://www.figma.com/file/uRV3Nfowdzt5CknW1FgUkQ/%F0%9F%93%9A-Fractal-Design-System-%7C-Base-Components-%7C-October-2022?node-id=1090%3A11670 for heights that you may encounter

Da-Colon commented on src/components/ui/Sidebar/index.tsx:
```diff
@@ -0,0 +1,157 @@
+        <SidebarTooltipWrapper label="Activity Feed">
+          <Link
+            data-testid="sidebar-activityLink"
+            to={DAO_ROUTES.activties('0x')}
```
Introduction a new way to track routes. see src/routes/constants

herbig replied:
@decent-dao/fractal lots of helpful comments from David in this PR for understanding the refactor

Da-Colon commented on src/routes/constants/dao.ts:
```diff
@@ -0,0 +1,9 @@
+export const DAO_ROUTES = {
```
In a follow PR I'll be using these values to update the different <Route> components keeping the routes defined in one location

Da-Colon reviewed (commented):
I need to check for/add translations for the tooltip labels. Will push up a commit later

herbig commented on src/components/ui/Sidebar/index.tsx:
```diff
@@ -0,0 +1,159 @@
+            target="_blank"
+          >
+            <SupportQuestion
+              aria-label="Support External Link"
```
I think we can add these to translation files as well right?

Da-Colon replied:
wasn't sure if they were user facing or not and need it. but yeah I can get that updated

Da-Colon replied:
I may have over used them. here. after reviewing the documentation. I'll take another look at these before these are merged in. if any remain I'll add them to the translation file

herbig replied:
awesome

herbig commented on src/components/ui/Sidebar/index.tsx:
```diff
@@ -0,0 +1,159 @@
+            target="_blank"
+          >
+            <Discord
+              aria-label="Discord Link"
```
same for this

herbig reviewed (changes requested):
@Da-Colon this looks good to me, except I believe we can add translations for `aria-label`, but I might be wrong.

ALSO the sidebar looks great in Chrome (and Safari but cool), but isn't so hot in Firefox, so some adjustments might be needed there.

ALSO ALSO, I'm assuming the proper fonts will come later as well?

herbig commented:
@Da-Colon and _Favorites_ should be baseline with _Connect Wallet_, but that might be a problem with the fonts not being in place

herbig commented:
@Da-Colon last thing, what's this vertical line in your screenshot, I don't see it in the preview link:

<img width="159" alt="Screen Shot 2022-10-20 at 12 32 10 PM" src="https://user-images.githubusercontent.com/384559/197006891-134763e3-bed0-429f-bbd8-f4f1b7409318.png">

herbig commented:
> > ALSO ALSO, I'm assuming the proper fonts will come later as well?
> 
> Fonts are in place for wallet menu and sidebar. The DAO Tree and Favorites in the header are just placeholders until design on it is finalized

That's the proper font for "Connect Wallet"?

Da-Colon commented:
technically there isn't a design for this. but I will switch it to Mono to match clickable things.

herbig commented:
> technically there isn't a design for this. but I will switch it to Mono to match clickable things.

I think it should match what's in this, and the wallet address and Disconnect should be updated as well

<img width="424" alt="Screen Shot 2022-10-20 at 12 41 27 PM" src="https://user-images.githubusercontent.com/384559/197010283-2d843a79-9ea7-4081-a17a-c901cc75be24.png">

herbig commented:
> > technically there isn't a design for this. but I will switch it to Mono to match clickable things.
> 
> I think it should match what's in this, and the wallet address and Disconnect should be updated as well
> 
> <img alt="Screen Shot 2022-10-20 at 12 41 27 PM" width="424" src="https://user-images.githubusercontent.com/384559/197010283-2d843a79-9ea7-4081-a17a-c901cc75be24.png">

Plus _Wallet_ and _Network_ titles

Da-Colon commented:
![image](https://user-images.githubusercontent.com/38386583/197011973-58cae748-73d2-46c6-b52a-9e03961a0090.png)

Da-Colon commented:
> @Da-Colon last thing, what's this vertical line in your screenshot, I don't see it in the preview link:
> 
> <img alt="Screen Shot 2022-10-20 at 12 32 10 PM" width="159" src="https://user-images.githubusercontent.com/384559/197006891-134763e3-bed0-429f-bbd8-f4f1b7409318.png">

This may just be my monitor and screen shot tool. isn't seen in app

herbig commented:
> ```ts
>         <Text
>           textStyle="text-sm-sans-regular"
>           color="chocolate.100"
>         >
>           {t('network')}
>         </Text>
> ```
> 
> ![image](https://user-images.githubusercontent.com/38386583/197011412-ef942f28-ab2f-4354-958b-b0122e27ec0d.png)
> 
> The font matches. and these new states were just added this morning and will have to wait for another PR as its out of scope of this one.

Agreed that it's out of scope for this PR, but something must be up if the font is set to that, because I can see from this screenshot that it's a serif font:

<img width="264" alt="Screen Shot 2022-10-20 at 1 05 15 PM" src="https://user-images.githubusercontent.com/384559/197013885-9f47b0a8-a03b-4c69-9570-f4bb0a1cf751.png">

Da-Colon commented:
So there does appear to be an issue with the font loading. I will follow up on this in a future PR. (looks like i have some hover states to update as well)

Da-Colon commented:
Icons are fixed (updated release to 0.0.8), Also updated the `aria-labels`

TL:DR. They are used when you have an interact-able elements that doesn't contain text. In this case <a> and <Link> elements. for screen readers. I've updated these values  and placed them on the <a> and <Link> elements and added the text  to the translations

herbig reviewed (approved):
Nice, looking good on Firefox, as discussed the objectively best browser.
